src/datasources/blockchain/blockchain-api.manager.ts

```diff
@@ -114,7 +110,7 @@ export class BlockchainApiManager implements IBlockchainApiManager {
 
       await this.cacheService.hSet(
         cacheDir,
-        result,
+        typeof result === 'string' ? result : JSON.stringify(result),
```

Comment:
Is this behavior difference needed? If the response is not a string (`eth_getBlockByNumber` returns an object) then it get stringfied, but the method is returning the cache key plain content in the line 96 (without a `JSON.parse` call). While this works for strings, it wouldn't work for other objects if I'm not wrong.

Have you considered using `JSON.stringify` and `JSON.parse` regardless of the value type?

Reply:
Good point. Changed in 494f05e15b485ee017aa31dc3d119d7a407138e3.